modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp

```diff
@@ -18,26 +18,33 @@ String paramName = baseAddressCheckoutStepDisplayContext.getParamName();
 
 long commerceAddressId = BeanParamUtil.getLong(baseAddressCheckoutStepDisplayContext.getCommerceOrder(), request, paramName);
 
+boolean validCommerceAddressId = false;
+
 CommerceAddress orderCommerceAddress = baseAddressCheckoutStepDisplayContext.getCommerceAddress(commerceAddressId);
 
 if ((orderCommerceAddress == null) || (orderCommerceAddress.getClassNameId() != PortalUtil.getClassNameId(CommerceOrder.class))) {
 	commerceAddressId = baseAddressCheckoutStepDisplayContext.getDefaultCommerceAddressId(commerceContext.getCommerceChannelId());
 
-	boolean validCommerceAddressId = false;
-
-	if (!validCommerceAddressId) {
-		for (CommerceAddress validCommerceAddress : commerceAddresses) {
-			if (commerceAddressId == validCommerceAddress.getCommerceAddressId()) {
-				validCommerceAddressId = true;
-			}
+	for (CommerceAddress validCommerceAddress : commerceAddresses) {
+		if (commerceAddressId == validCommerceAddress.getCommerceAddressId()) {
+			validCommerceAddressId = true;
 		}
 	}
+}
+else {
+	for (CommerceAddress validCommerceAddress : commerceAddresses) {
+		if (Objects.equals(orderCommerceAddress.getName(), validCommerceAddress.getName()) && Objects.equals(orderCommerceAddress.getStreet1(), validCommerceAddress.getStreet1()) && Objects.equals(orderCommerceAddress.getStreet2(), validCommerceAddress.getStreet2()) && Objects.equals(orderCommerceAddress.getStreet3(), validCommerceAddress.getStreet3()) && (orderCommerceAddress.getZip() == validCommerceAddress.getZip()) && (orderCommerceAddress.getCountryId() == validCommerceAddress.getCountryId()) && (orderCommerceAddress.getRegionId() == validCommerceAddress.getRegionId()) && (orderCommerceAddress.getType() == validCommerceAddress.getType()) && (orderCommerceAddress.getLatitude() == validCommerceAddress.getLatitude()) && (orderCommerceAddress.getLongitude() == validCommerceAddress.getLongitude())) {
```

Comment:
can't we use Objects.equals(validCommerceAddress, orderCommerceAddress) ? instead of checking every field? or is there the possibility that the addresses we are checking are somehow different but still valid?